File: algorithms/gcd-lcm/GcdLcm.scala
import scala.annotation.tailrec

@tailrec
def gcd(a:Long,b:Long) : Long = if (b != 0) gcd(b,a % b) else a

def gcdList(a:Long*) : Long = a.reduce((a,b) => gcd(a,b))

gcdList(42,231,105)

File: algorithms/gcd-lcm/GcdLcm.py
import functools
from typing import List

def gcd(a:int, b:int) -> int:
  if b!=0:
    return gcd(b,a%b)
  else:
    return a


def lcm(a:int, b:int) -> int : return a * b // gcd(a,b)


def gcd_list(xlist : List[int]) -> int: return functools.reduce(lambda a,b: gcd(a,b), xlist)


def lcm_list(xlist : List[int]) -> int: return functools.reduce(lambda a,b: lcm(a,b), xlist)




gcd_list([42,231,105])

